tests/syscall/consolefs/enc/enc.c

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/internal/tests.h>
+#include <sys/ioctl.h>
+#include <termios.h>
+#include <unistd.h>
+#include "consolefs_t.h"
+
+static bool _termios_flags_equal(
+    const struct termios* a,
+    const struct termios* b)
+{
+    return a->c_iflag == b->c_iflag && a->c_oflag == b->c_oflag &&
+           a->c_cflag == b->c_cflag && a->c_lflag == b->c_lflag;
+}
+
+void test_consolefs()
+{
+    // test ioctl
+    struct termios t1 = {0};
+    struct termios t2 = {0};
+    OE_TEST(ioctl(STDIN_FILENO, TCGETS, &t1) == 0);
+    OE_TEST(!_termios_flags_equal(&t1, &t2));
+    OE_TEST(ioctl(STDIN_FILENO, TCSETS, &t1) == 0);
+    OE_TEST(ioctl(STDIN_FILENO, TCGETS, &t2) == 0);
+    OE_TEST(_termios_flags_equal(&t1, &t2));
```

Comment:
The test fails because the container runs without a tty. What do you think about this change:
```suggestion
    const int res = ioctl(STDIN_FILENO, TCGETS, &t1);
    if (res == -1)
        // expected to fail if there's no tty (e.g., when run in a container)
        OE_TEST(errno == ENOTTY);
    else
    {
        struct termios t2 = {0};
        OE_TEST(!_termios_flags_equal(&t1, &t2));
        OE_TEST(ioctl(STDIN_FILENO, TCSETS, &t1) == 0);
        OE_TEST(ioctl(STDIN_FILENO, TCGETS, &t2) == 0);
        OE_TEST(_termios_flags_equal(&t1, &t2));
    }
```

Reply:
Consider using musl provided `isatty()`.

Reply:
Good idea, but I tried it and it doesn't work because musl's implementation uses the TIOCGWINSZ ioctl. The OE implementation handles it inside the enclave and always succeeds, so `isatty` always returns 1.